### ISSUE TITLE
doxygen does not correctly render link content in MarkDown file: expected <code>...</code>

### ISSUE THREAD

**Issue by GerHobbelt** [doxygen/doxygen#10475]
**Describe the bug**

Running `doxygen.exe Doxyfile` (MSWindows 10, 64-bit) against a tree of MarkDown + source files.
Assuming #10473 is fixed (or the EXCLUDE_PATTERNS line in the demo's Doxyfile config file **uncommented**), the markdown files are rendered.

However, the input (README.md):

```
(line 9) This is some text with a link to [`doc/OTHER.md` file](doc/OTHER.md) using MarkDown link/reference notation.
```

renders this HTML:

```
<p>(line 9) This is some text with a link to `doc/OTHER.md` file using MarkDown link/reference notation.</p>
```

Note the `` `\doc/OTHER.md\` `` which was part of the link text is not converted, as expected, to `<code>doc/OTHER.md</code>`.

Possibly this related to the failure to render the `<a href...>` link as well, as mentioned in #10474?

**To Reproduce**

Clone/copy the repo https://github.com/GerHobbelt/doxygen-markdown-linking-problem and run `doxygen.exe Doxyfile` in the root of that repo dir tree (so relative paths SHOULD be okay); be sure to *edit* the `Doxyfile` to **uncomment** Doxyfile::line 12 (`EXCLUDE_PATTERNS       = */uber-long-file-path5/*`) or you'll trigger #10473 before you're able to see this bug in the generated output.

**Version**

Tested to fail as described when using the latest Win64 build artifact (doxygen.exe):

```
$ doxygen.exe -v
1.10.0 (06fcb408e9de97cc6acfdbec4e261a302c0d204b*)
```

Probably related issues:

- #10473
- #10474

**Comment by GerHobbelt** [comment]
*tweaked* & cleaned zip demo again:

[buggered-doxygen.zip](https://github.com/doxygen/doxygen/files/13628660/buggered-doxygen.zip)

For  an analysis, expectations and my comments, see the message: https://github.com/doxygen/doxygen/issues/10474#issuecomment-1849027662-permalink, *in particular the "Asides" mentioned there*.

**Comment by doxygen** [comment]
@GerHobbelt Please verify if the reference commit fixes the problem for you. Do not close the issue, this will be done automatically when the next release is made available.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
